packages/service-library/src/web-api/web-api-error-codes.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// tslint:disable: no-unnecessary-class variable-name
+
+export declare type WebApiErrorCodeName =
+    | 'ResourceNotFound'
+    | 'InvalidResourceId'
+    | 'InvalidJsonDocument'
+    | 'RequestBodyTooLarge'
+    | 'InvalidURL'
+    | 'InternalError'
+    | 'MissingApiVersionQueryParameter'
+    | 'MissingContentTypeHeader'
+    | 'UnsupportedContentType'
+    | 'UnsupportedApiVersion';
+
+export interface WebApiErrorCode {
+    statusCode: number;
+    response: {
+        code: WebApiErrorCodeName;
+        error: string;
+    };
+}
+
+export class WebApiErrorCodes {
+    public static resourceNotFound: WebApiErrorCode = {
+        statusCode: 404,
+        response: {
+            code: 'ResourceNotFound',
+            error: 'The specified resource does not exist.',
+        },
+    };
+
+    public static invalidResourceId: WebApiErrorCode = {
+        statusCode: 400,
+        response: {
+            code: 'InvalidResourceId',
+            error: 'The resource ID is not valid.',
+        },
+    };
+
+    public static invalidURL: WebApiErrorCode = {
+        statusCode: 400,
```

Comment:
do you also want to add http status code for this?
what about batch requests? you can have 1 url that is not valid, in that case are we planning to reject the whole request or just for the invalid ones? same applies for the internal error error code.

Reply:
In a batch we will return scan result object in array w/o rejecting request